tests/components/pi_hole/test_init.py

```diff
@@ -127,37 +167,44 @@ async def test_enable_service_call(hass):
             {
                 pi_hole.DOMAIN: [
                     {"host": "pi.hole1", "api_key": "1"},
-                    {"host": "pi.hole2", "name": "Custom", "api_key": "2"},
+                    {"host": "pi.hole2", "name": "Custom"},
                 ]
             },
         )
 
         await hass.async_block_till_done()
 
         await hass.services.async_call(
-            pi_hole.DOMAIN, pi_hole.SERVICE_ENABLE, {}, blocking=True
+            pi_hole.DOMAIN,
+            SERVICE_DISABLE,
+            {ATTR_ENTITY_ID: "all", SERVICE_DISABLE_ATTR_DURATION: "00:00:01"},
+            blocking=True,
         )
 
         await hass.async_block_till_done()
 
-        assert mocked_hole.enable.call_count == 2
+        mocked_hole.disable.assert_called_once_with(1)
 
 
-async def test_update_coordinator(hass):
-    """Test update coordinator."""
+async def test_unload(hass):
+    """Test unload entities."""
+    entry = MockConfigEntry(
+        domain=pi_hole.DOMAIN,
+        data={
+            CONF_NAME: DEFAULT_NAME,
+            CONF_HOST: "pi.hole",
+            CONF_LOCATION: DEFAULT_LOCATION,
+            CONF_SSL: DEFAULT_SSL,
+            CONF_VERIFY_SSL: DEFAULT_VERIFY_SSL,
+        },
+    )
+    entry.add_to_hass(hass)
     mocked_hole = _create_mocked_hole()
-    sensor_entity_id = "sensor.pi_hole_ads_blocked_today"
     with _patch_config_flow_hole(mocked_hole), _patch_init_hole(mocked_hole):
-        assert await async_setup_component(
-            hass, pi_hole.DOMAIN, {pi_hole.DOMAIN: [{"host": "pi.hole"}]}
-        )
+        await hass.config_entries.async_setup(entry.entry_id)
         await hass.async_block_till_done()
-        assert mocked_hole.get_data.call_count == 3
-        assert hass.states.get(sensor_entity_id).state == "0"
+    assert entry.entry_id in hass.data[pi_hole.DOMAIN]
 
-        mocked_hole.data["ads_blocked_today"] = 1
-        utcnow = dt_util.utcnow()
-        async_fire_time_changed(hass, utcnow + MIN_TIME_BETWEEN_UPDATES)
-        await hass.async_block_till_done()
-        assert mocked_hole.get_data.call_count == 4
-        assert hass.states.get(sensor_entity_id).state == "1"
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    await hass.async_block_till_done()
+    assert entry.entry_id not in hass.data[pi_hole.DOMAIN]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)